packages/core-transaction-pool/lib/guard.js

```diff
@@ -165,15 +165,28 @@ module.exports = class TransactionGuard {
    */
   __determineValidTransactions () {
     this.transactions.forEach(transaction => {
-      if (transaction.type === TRANSACTION_TYPES.TRANSFER) {
+      switch (transaction.type) {
+      case TRANSACTION_TYPES.TRANSFER:
         if (!isRecipientOnActiveNetwork(transaction)) {
           this.__pushError(transaction, `Recipient ${transaction.recipientId} is not on the same network: ${configManager.get('pubKeyHash')}`)
           return
         }
-      }
-
-      if (this.pool.checkIfSenderHasVoteTransactions(transaction.senderPublicKey)) {
-        this.__pushError(transaction, `Sender ${transaction.senderPublicKey} already has a pending vote transaction in pool`)
+        break
+      case TRANSACTION_TYPES.VOTE:
+        if (this.pool.checkIfSenderHasVoteTransactions(transaction.senderPublicKey)) {
+          this.__pushError(transaction, `Sender ${transaction.senderPublicKey} already has a pending vote transaction in pool`)
+          return
+        }
+        break
+      case TRANSACTION_TYPES.SECOND_SIGNATURE:
```

Comment:
`default` should be sufficient if none of the other types are handled in another case.

Reply:
Yes, technically `default:` suffices, but I added them in order to be explicit at what is being invalidated. I presume nobody remembers all transaction types on top of his head, so reading them here helps to skip looking up the definition of `TRANSACTION_TYPES`.